bodyswap.lua

```diff
@@ -33,15 +33,58 @@ function clearNemesisFromLinkedSites(nem)
     end
 end
 
-function createNemesis(unit)
+-- shamelessly copypasted from makeown.lua
```

Comment:
You can already call the `name_unit` function from `makeown`. is that sufficient?
```lua
local makeown = reqscript('makeown')

makeown.name_unit(unit)
```

Reply:
Not exactly! Makeown makes a Dwarven name if no valid race language is found. But I wanted to replicate the wild beast behaviour when a beast becomes a historical figure, or a night creature, and I believe a divine name is used

Additionally, in adv mode the name that matters is the historical figure one, the unit name doesn't seem to be used

Reply:
should `makeown` follow that behavior? the hist fig name *should* be synced with the unit name, and if `makeown` isn't doing that, it probably should be. I don't know about the divine name vs dwarven name, though. that might be useful to parameterize. what kinds of names do dwarven pets get in fort mode? what about named war beasts brought by foreign civilizations?

Reply:
yeah, looking at `makeown`, I think this logic should be moved there.

Reply:
bumping this comment. this is way too much logic to get copied around